icu4j/main/classes/core/src/com/ibm/icu/impl/Punycode.java

```diff
@@ -124,6 +124,12 @@ public static StringBuilder encode(CharSequence src, boolean[] caseFlags) throws
         int n, delta, handledCPCount, basicLength, bias, j, m, q, k, t, srcCPCount;
         char c, c2;
         int srcLength = src.length();
+        if (srcLength > 1000) {
```

Comment:
Define this 1000 as a name private static . Also, why such check is not in the C++ version?

Reply:
1. ditto
2. see the PR description and last ticket comment